certbot/tests/util_test.py

```diff
@@ -277,15 +280,17 @@ def test_multiple(self):
         for f, _ in items:
             f.close()
 
+    @mock.patch("certbot.util.os.open")  # Avoid orphan file descriptor
     @mock.patch("certbot.util.os.fdopen")
-    def test_failure(self, mock_fdopen):
+    def test_failure(self, mock_fdopen, _):
         err = OSError("whoops")
         err.errno = errno.EIO
         mock_fdopen.side_effect = err
         self.assertRaises(OSError, self._call, "wow")
 
+    @mock.patch("certbot.util.os.open")  # Avoid orphan file descriptor
     @mock.patch("certbot.util.os.fdopen")
-    def test_subsequent_failure(self, mock_fdopen):
+    def test_subsequent_failure(self, mock_fdopen, _):
```

Comment:
I'd appreciate a sanity check, but I feel like it's more realistic these errors come from `os.open` rather than `os.fdopen` anyway. If you agree, how about we just mock `os.open` and raise the exception there?

Reply:
See the other similar issue.